lib/semantics/check-omp-structure.cc

```diff
@@ -238,12 +206,51 @@ void OmpStructureChecker::Enter(const parser::OmpBlockDirective &x) {
         OmpClause::IF, OmpClause::NUM_THREADS, OmpClause::PROC_BIND};
     SetContextAllowedOnce(allowedOnce);
   } break;
+  // 2.7.3 single-clause -> private-clause |
+  //                        firstprivate-clause
+  case parser::OmpBlockDirective::Directive::Single: {
+    SetContextDirectiveEnum(OmpDirective::SINGLE);
+    OmpClauseSet allowed{OmpClause::PRIVATE, OmpClause::FIRSTPRIVATE};
+    SetContextAllowed(allowed);
```

Comment:
If you replaced these two lines with `SetContextAllowed({OmpClause::PRIVATE, OmpClause::FIRSTPRIVATE});` you wouldn't need the block around the body of the `case`.